src/item/enchantment/AvailableEnchantmentRegistry.php

```diff
@@ -149,35 +149,42 @@ public function setSecondaryItemTags(Enchantment $enchantment, array $tags) : vo
 	}
 
 	/**
-	 * Returns enchantments that can be applied to the specified item.
-	 *
-	 * @param bool $onlyPrimaryTags Whether to check only primary item tags for each enchantment.
-	 *                              This should only be used to select available enchantments for an enchanting table.
+	 * Returns enchantments that can be applied to the specified item in an enchanting table.
 	 *
 	 * @return Enchantment[]
 	 */
-	public function getAvailableEnchantments(Item $item, bool $onlyPrimaryTags = false) : array{
+	public function getAvailableEnchantingTableEnchantments(Item $item) : array{
 		if(count($item->getEnchantmentTags()) === 0){
 			return [];
 		}
 
 		return array_filter(
 			$this->enchantments,
-			fn(Enchantment $enchantment) => $this->isAvailableForItem($enchantment, $item, $onlyPrimaryTags)
+			fn(Enchantment $e) => TagRegistry::getInstance()->isTagArraySubset($this->getPrimaryItemTags($e), $item->getEnchantmentTags())
 		);
 	}
 
 	/**
-	 * Returns whether the specified enchantment can be applied to the particular item.
+	 * Returns enchantments that can be applied to the specified item.
 	 *
-	 * @param bool $onlyPrimaryTags Whether to check only primary item tags for the enchantment.
-	 *                              This should only be used to select available enchantments for an enchanting table.
+	 * @return Enchantment[]
 	 */
-	public function isAvailableForItem(Enchantment $enchantment, Item $item, bool $onlyPrimaryTags = false) : bool{
-		$enchantmentTags = $this->getPrimaryItemTags($enchantment);
-		if(!$onlyPrimaryTags){
-			$enchantmentTags = array_merge($enchantmentTags, $this->getSecondaryItemTags($enchantment));
+	public function getAvailableEnchantments(Item $item) : array{
+		if(count($item->getEnchantmentTags()) === 0){
+			return [];
 		}
+
+		return array_filter(
+			$this->enchantments,
+			fn(Enchantment $enchantment) => $this->isAvailableForItem($enchantment, $item)
+		);
+	}
+
+	/**
+	 * Returns whether the specified enchantment can be applied to the particular item.
+	 */
+	public function isAvailableForItem(Enchantment $enchantment, Item $item) : bool{
```

Comment:
Is there a reason to expose this publicly?

Reply:
Yes. This method will be needed in the future to test the availability of enchantments separately in anvil